test/master_gateway.js

```diff
@@ -77,6 +77,32 @@ describe('Master gateway instantiation', function() {
         assert.equal(instance.slaves.hasOwnProperty('dummy'), false);
     });
 
+    it('Should prevent another loaded callback if a slave is already loaded', function() {
+        instance.addSlave({
+            frameId : 'dummy',
+            slaveId : 'dummy',
+        });
+        instance.slaves['dummy'].isLoaded = true;
```

Comment:
You should call `instance.slaveLoaded({}, slaveData);` in this line too. This is cheating :D

Reply:
@dikaso Nice catch :D